### PR TITLE
feat: Adds prometheus alert rules

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Adds Prometheus and ~Loki~ alert rules following spec TE071.

Initially this PR included Loki rules, but because of[ this bug ](https://github.com/canonical/loki-k8s-operator/issues/403)those rules have been removed and should be addressed when the bug is fixed.

To test deploy the charm from this branch and follow the [guide on how to integrate with COS](https://charmhub.io/vault-k8s/docs/h-integrate-with-cos), go to the alerts tabs in Grafana and then:
- To test the vault sealed rule, seal and unseal vault and observe the alert firing
- To test the leader last contact rule, reduce the threshold to something very small, maybe 5 ms and observe the alert firing
- To test the failure tolerance rule scale vault up to 3 units then back to 2 units and observe the alert firing
- I couldn't find a way to simulate leadership setup failure

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/prometheus_alert_rules/vault_sealed.rule [hunk 1]
```diff
@@ -0,0 +1,8 @@
+alert: VaultSealedRule
+expr: vault_core_unsealed == 0
+for: 0m
+labels:
+    severity: critical
+annotations:
+    summary: Vault has gone sealed again (instance {{ $labels.instance }})
```
Comment (gruyaume) [inline src/prometheus_alert_rules/vault_sealed.rule:7]: I'm not sure about the phrasing of `has gone sealed again`. First, `has gone sealed` is strange, it's more that `Vault is sealed`. And the `again` can also not be true, on first startup, Vault will be sealed and it won't be "again".

File: src/charm.py
Comment (gruyaume) [inline src/charm.py:1]: Have we tested the alter rules and they work fine?

File: src/prometheus_alert_rules/vault_sealed.rule (hunk 1 above)
Comment (saltiyazan) [reply src/prometheus_alert_rules/vault_sealed.rule:7]: Improved

**Review by ghislainbourgeois — APPROVED** [review]
Approved. Your branch is out-of-date, please rebase or merge main on it before merging.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**